packages/dds/merge-tree/src/test/mergeTreeOperationRunner.ts

```diff
@@ -360,11 +361,24 @@ export function generateOperationMessagesForClients(
 		let op: IMergeTreeOp | undefined;
 		if (len === 0 || len < minLength) {
 			const text = client.longClientId!.repeat(random.integer(1, 3));
-			op = client.insertTextLocal(random.integer(0, len), text);
+			let pos = random.integer(0, len);
+			const fieldEndpoints = posInField(client, pos);
+
+			if (fieldEndpoints !== undefined) {
+				pos = 0;
+			}
+			op = client.insertTextLocal(pos, text);
 		} else {
 			let opIndex = random.integer(0, operations.length - 1);
-			const start = random.integer(0, len - 1);
-			const end = random.integer(start + 1, len);
+			let start = random.integer(0, len - 1);
```

Comment:
just leave this unchanged

Reply:
I left a TODO in the code, but without checking the endpoints here, one of the seeds hits a `MergeTree insert failed` error.

Reply:
i still don't want this to change. the operations you use can totally ignore the start and end values that we generate here if they want, so i don't see why the code here has any specific impact